src/Helper.php

```diff
@@ -114,10 +114,47 @@ public static function get_shipment_carriers() {
      * @return array List of aliases.
      */
     public static function get_aliases() {
-        return [
-            'delivered' => __( 'Delivered', 'trackmage' ),
-            'shipped'   => __( 'Shipped', 'trackamge' ),
-        ];
+        $workspaceId = get_option( 'trackmage_workspace', 0 );
+        $cachedAliases = get_transient('trackmage_order_statuses');
+        $aliases = [];
+        if($cachedAliases && isset($cachedAliases[$workspaceId])) {
+            $aliases = $cachedAliases[$workspaceId];
+        }elseif($workspaceId !== 0){
+            try {
+                $client = Plugin::get_client();
+                $guzzleClient = $client->getGuzzleClient();
+                $response = $guzzleClient->get("/workspaces/{$workspaceId}/order_statuses");
+                $content = $response->getBody()->getContents();
+                $data = json_decode($content, true);
+
+                foreach ($data['hydra:member'] as $status) {
+                    $name = $status['name'];
+                    $aliases[$name] = __(ucfirst($name),'trackmage');
+                }
+
+                $cachedAliases[$workspaceId] = $aliases;
+                set_transient('trackmage_order_statuses', $cachedAliases, 3600);
+            } catch( ApiException $e ) {
+
+            }
+
+        }
+        return $aliases;
+    }
+
+    /**
+     * Returns the used aliases.
+     *
+     * @return array List of aliases.
+     */
+
+    public static function get_used_aliases() {
+        $usedAliases = get_option( 'trackmage_order_status_aliases', [] );
```

Comment:
Can't we use just `return array_values(get_option( 'trackmage_order_status_aliases', [] ))` here?

Also, when user changes workspace (in Trackmage Settings page) we should drop the value of trackmage_order_status_aliases

Reply:
@karser We can if we change edit/delete status code. Current code stores empty alias for statuses in  'trackmage_order_status_aliases' . So we can modify code to don't store empty aliases. If it's OK.

Reply:
array_values removes falsy values, e.g `array_values(['', null, 0, '0']) === []`

Reply:
@karser 
![image](https://user-images.githubusercontent.com/5622709/66768620-aa999e00-eebb-11e9-884b-4d5052f06d7b.png)

result:

![image](https://user-images.githubusercontent.com/5622709/66768724-f0566680-eebb-11e9-94b3-cca03b273c94.png)

Reply:
Ah, sorry. That's `array_filter(array_values(['', null, 0, '0']))`.  :)

Reply:
@karser so do you think those two functions will work faster than one foreach?

Reply:
`array_values(array_filter())` is [6x faster](http://sandbox.onlinephpfunctions.com/code/ae0125ca7cceec0ae713a22f3db3d27fc716bf88) and is more readable.
![image](https://user-images.githubusercontent.com/1675033/66772466-79be6680-eec5-11e9-9c45-b09d14641b8a.png)